File: plugin/src/org/jetbrains/yesod/julius/highlight/JuliusAnnotator.java
package org.jetbrains.yesod.julius.highlight;

/**
 * @author Leyla H
 */

import com.intellij.lang.annotation.AnnotationHolder;
import com.intellij.lang.annotation.Annotator;
import com.intellij.openapi.editor.DefaultLanguageHighlighterColors;
import com.intellij.psi.PsiElement;
import org.jetbrains.yesod.julius.psi.*;
import org.jetbrains.yesod.julius.psi.String;

public class JuliusAnnotator implements Annotator {
    @Override
    public void annotate(PsiElement psiElement, AnnotationHolder annotationHolder) {

        if(psiElement instanceof Operator) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(JuliusColors.OPERATOR);
        }
        if(psiElement instanceof Comments) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(JuliusColors.COMMENTS);
        }
        if(psiElement instanceof Curly) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(JuliusColors.TEXT);
        }
        if(psiElement instanceof Sign) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(JuliusColors.SIGN);
        }
        if(psiElement instanceof String) {
            annotationHolder.createInfoAnnotation(psiElement, null).setTextAttributes(JuliusColors.STRING);
        }
    }
}

File: plugin/src/org/jetbrains/yesod/julius/highlight/JuliusAnnotator.kt
package org.jetbrains.yesod.julius.highlight

/**
 * @author Leyla H
 */

import com.intellij.lang.annotation.AnnotationHolder
import com.intellij.lang.annotation.Annotator
import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
import com.intellij.psi.PsiElement
import org.jetbrains.yesod.julius.psi.*
import org.jetbrains.yesod.julius.psi.String
import org.jetbrains.yesod.julius.psi.Number

class JuliusAnnotator : Annotator {
    override fun annotate(psiElement: PsiElement, annotationHolder: AnnotationHolder) {

        if (psiElement is Keyword) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = DefaultLanguageHighlighterColors.KEYWORD
        }
        if (psiElement is String) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = JuliusColors.STRING
        }
        if (psiElement is Comment) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = JuliusColors.COMMENT
        }
        if (psiElement is DotIdentifier) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = JuliusColors.DOT_IDENTIFIER
        }
        if (psiElement is Number) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = JuliusColors.NUMBER
        }
        if (psiElement is Interpolation) {
            annotationHolder.createInfoAnnotation(psiElement, null).textAttributes = JuliusColors.INTERPOLATION
        }
    }
}